### DISCUSSION TITLE
Wiz Bulb Power Consumption

### DISCUSSION THREAD

**Discussion by zerrium** [sbidy/wiz_light#165] [Q&A]
Is there anyway to measure Wiz bulb power consumption on Home Assistant? No sensor yet?

**Comment by sbidy** [comment]
I don't see any sensor or other data point for monitoring or measuring the power consumption of the bulb.

**Comment by danielr1996** [comment]
Do you want to measure the power consumption in real time or are you just interrested in the general consumption in on/off state. I think the power consumption should not really change depending on the color or effects so theorectically you could just use the on/off times which should be available through home assistant and the power consumption for on/off state and calculate the consumption over a given period yourself. If you really need the current consumption you could use a shelly 1PM and install it behind your switch to measure the real time power consumption, howerver given your budget and knowledge in electrical installation this might not be feasible.

**Reply by danielr1996** [reply]
Ok so grabed a cheap power measurement device from amazon and tested a few scenarios, in idle an A60 E27 bulb consumes around 0.4W, in cold white around 7W, in warm white around 4W. But if can extract the color temp, rgb color and brightness from HA you might still be able to estimate the power consumption, but I dont know if thats worth the struggle.

**Comment by abigdeel** [comment]
The Wiz app itself does have a tab for power consumption. I'm assuming they've already done this work and created a relationship for color/brightness and power.

I doubt there's any easy way to pull the data, but it is already there. It would be great if Wiz could just share that and also add support for transition time for the lights :)

**Reply by danielr1996** [reply]
Oh thats interesting, I tought the power consumption tab was for the smart plugs, but indeed it shows the consumption for my bulbs.

**Reply by abigdeel** [reply]
I was thinking about it and it wouldn't be too much work to figure out the brightness, color, power relationship by using that power consumption tab. If you're really feeling up to it lol.

- Just go from 1% brightness up to 100% in like 10 increments and note the power.
- Then repeat that again a handful of times at different colors.
- Also use as many lights as possible, since the power consumption tab rounds to the nearest whole watt. But the app does add together individual light power consumption with decimal point precision (i.e. my room lights are showing 4W and 4W, but the total for the room is 9W).

And then do some multivariable regression in Excel or something with that data.

**Reply by danielr1996** [reply]
If I can find out how to intercept network calls from my android phone I will try to figure out how the data is interchanged between the bulb and app, I'm afraid that it's not as easy as sending commands to the bulb, but let's see. If it's indeed UDP then it should be easy to integrate it at least into the underlaying library pywizlight, for the integration part I don't know what is to be done to provide power consumption to home assistant.

**Reply by sbidy** [reply]
Hey,
the bulb "API" have more properties and functions (e.g. the power consumption, group etc.). But this API is called via MQTT messages from the WiZ Cloud services. The app will use this service.
The connection (MQTT via TLS) between the bulb and the cloud is encrypted, and the bulb will be authenticated via certificate.

A possible way is to (and the most common on) communicate with the WiZ Cloud Rest API to interact with the bulbs. This will be done e.gg by the Apps.

This implementation (wiz_light) dont use the MQTT or Cloud interfaces. All calls going directly to the bulb via UPD calls. And this UDP "API" is very limited (and also not that reliable).

My hope is, that WiZ will also provide the MQTT endpoint to the public so that we can use the MQTT integration from HASS to control the bulb in the same way as the cloud does.

Hope this makes some points a bit clearer :-)

**Reply by danielr1996** [reply]
So I did a little test and unplugged my router and the power consumption tab showed "no data available", so I suppose this works with the cloud

**Reply by ScottG489** [reply]
Just to be clear, we're still saying that this is possible, correct? As stated before, all the power consumption is likely calculated based on brightness, temperature, and color. So calculating the power consumption is something that could be done locally using information we already have.

**Reply by brettonw** [reply]
I have 12 bulbs in my house and the app says I used 3KWh for all of them last month. That's like... 1% of my household electric usage. I imagine if you are tracking the bulbs, you've got everything else in your house nailed down to the point that 1% error is significant? I'm not knocking the effort, just wondering if it's worth it? If you just guessed that all the bulbs are pulling 4W then you'd be off by no more than a few pennies on your electric cost...

**Reply by ScottG489** [reply]
That's a fair point. I looked around and I think this offers what I need:
https://github.com/bramstroker/homeassistant-powercalc

FWIW I still think adding this feature would add a nice polish to the integration, but I'm sure there are many other higher priorities.
